numpy/distutils/system_info.py

```diff
@@ -1724,26 +1732,26 @@ def has_cblas(self, info):
                 # check we can compile (find headers)
                 obj = c.compile([src], output_dir=tmpdir,
                                 include_dirs=self.get_include_dirs())
+            except (distutils.ccompiler.CompileError, distutils.ccompiler.LinkError):
+                return None
 
-                # check we can link (find library)
-                # some systems have separate cblas and blas libs. First
-                # check for cblas lib, and if not present check for blas lib.
+            # check we can link (find library)
+            # some systems have separate cblas and blas libs.
+            # We also check whether the supplied libraries contain the
+            # cblas interfaces
+            # Otherwise check for cblas lib, and if not present check for blas lib.
+            for libs in [info['libraries'], ['cblas'] + info['libraries'],
+                         ['blas'] + info['libraries'], ['cblas'], ['blas']]:
                 try:
                     c.link_executable(obj, os.path.join(tmpdir, "a.out"),
-                                      libraries=["cblas"],
+                                      libraries=libs,
                                       library_dirs=info['library_dirs'],
                                       extra_postargs=info.get('extra_link_args', []))
-                    res = "cblas"
+                    res = libs
+                    # This breaks the for loop
+                    break
```

Comment:
Just use `return libs` here

Reply:
done, this implies that the last `return res` also changes to `return None`, for clarity!